### ISSUE TITLE
Improve speed for initial sync with virtual files

### ISSUE THREAD

**Issue by wonx** [nextcloud/desktop#4424]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you want to have the same feature implemented.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Feature description
When using virtual files, the first log in after a new installation will start a syncing process that can take a very long time depending on the number of files to synchronize. 

In my case, i'm syncing around ~700000 files, my computer has been already up for 29 hours without a restart and the sync process has now reached the 50% mark. I can see that the virtual files are created one by one, but it can be as slow as 2 per second. Two or more days until Nextcloud can be usable is too much in my opinion.

It would be cool if there was any way to speed up the initial sync.

PS: This is related to https://github.com/nextcloud/desktop/issues/4421

**Comment by marcotrevisan** [comment]
I'm experiencing a similar problem on Mac OS (around 200k files). In my humble opinion, syncrhonizing the full hierarchy is the key problem here. The typical end user doesn't need to have the full folder hierarchy saved and synchronized. A lazier approach (i.e. trigger on open and/or scan the opened subfolders only, and not the whole depth trough but perhaps 1 or 2 levels below) would grant more scalability and decrease the load on the NC server.

**Comment by johannes-luebke** [comment]
I'd like to add, that restarting the sync, client or PC will result in a complete restart of the process. Also the sync doesn't seem to start immediately, but it first counts all files it will sync and then starts syncing. The counting alone takes two days for me and the sync isn't done after more than 10 days. At least the sync should pick up where it left of.

**Comment by CWempe** [comment]
I have the same issue.

The most annoying part is that I do not need the folders with all the little files available on my desktop.

So it would be enough if I could say "do not sync this folder unless it is accessed by the user".

I think the suggestion from @marcotrevisan (see https://github.com/nextcloud/desktop/issues/4464) also sounds promising.

**Comment by PhilippSchlesinger** [comment]
See https://github.com/nextcloud/desktop/issues/4918#issuecomment-1246386007 for a description of a problem with the tray window related to speed issues for inital sync

**Comment by PhilippSchlesinger** [comment]
With latest Nextcloud Client 3.7.3 an inital sync on ~150k files took <1 hour where it was a whole night and endless errors in the past.
Maybe you guys could also check again and see if it improved with the latest version.

**Comment by PhilippSchlesinger** [comment]
> With latest Nextcloud Client 3.7.3 an inital sync on ~150k files took <1 hour where it was a whole night and endless errors in the past.
> Maybe you guys could also check again and see if it improved with the latest version.

@CWempe 
Since you described the issue in detail and with numbers previously, could you maybe check again with 3.7.3 or later and report if anything changed?

**Comment by hodyroff** [comment]
@limatus Try with ownCloud Infinite Scale, 3.0 just got released, would expect 4x performance compared with oC10,

**Comment by limatus** [comment]
@hodyroff thank for the hint, but I do not intend to switch servers – the Server was and is from NC!

**Comment by tobiasKaminsky** [comment]
@claucambra is this a duplicate of [#5692](https://github.com/nextcloud/desktop/issues/5692 or vice vera?

**Comment by claucambra** [comment]
They are different, this is related to the Windows VFS (normal sync engine) while #5692 is related to the macOS-specific sync engine in the file provider module

**Comment by allexzander** [comment]
@limatus @CWempe Just to get a bit more context on Virtual Files vs normal sync, do you have a much slower syncing when using Virtual Files when compared to how it syncs via normal sync if you also select to sync everything?

**Comment by tomdereub** [comment]
@allexzander : The problem is only with the initial sync. I'm using VFS on my personal server with success, it's working well.
The problem appears with lots of data, with 500 000 files it takes about a few days to get the initial sync complete. After that syncing seems as quick as with normal sync.
Is there any chance to see any progress on this issue ? It has been agreed for 2 years now (https://github.com/nextcloud/desktop/issues/3120#issuecomment-907067592) without visible progress...

**Comment by limatus** [comment]
@allexzander if I sync the files via normal sync, the bottleneck seems to be the connection speed, which is understandable. Sadly, we mostly use virtual files, as they're simply too many files. It's similar to what @tomdereub mentioned, the initial sync needs days, thereafter, it’s fine.

**Comment by PhilippSchlesinger** [comment]
@allexzander for the sake of completeness I'd like to add that the @tomdereub and others [describe ](https://github.com/nextcloud/desktop/issues/4424#issuecomment-1663918953)also occurs if a relevant amount of files gets added to the nextcloud account *after* initial sync.
So if the nextcloud client has to sync that newly added amount of files, the client shows the same issue like on initial sync.

As described by @CWempe in https://github.com/nextcloud/desktop/issues/4424#issuecomment-1341235591, the syncing speed descreases dramatically over time.
Is this maybe due to listing activies in tray window in realtime for any single file that gets synced?
If this could be idientified as a root of slowdown, then maybe lazyloading activities or even summarized listing for huge amount of files would be an option.

**Comment by tomdereub** [comment]
> @allexzander : The problem is only with the initial sync. I'm using VFS on my personal server with success, it's working well. The problem appears with lots of data, with 500 000 files it takes about a few days to get the initial sync complete. After that syncing seems as quick as with normal sync. Is there any chance to see any progress on this issue ? It has been agreed for 2 years now ([#3120 (comment)](https://github.com/nextcloud/desktop/issues/3120#issuecomment-907067592)) without visible progress...

After some time using VFS on that folder with about 500 000 files, I find it too bad to keep syncing the whole folder tree. Every time somebody modifies quite a lot of files, it starts a long sync. It seems to me impossible to deploy for 30 persons, it will charge a lot the server and each computer.
From my point of view, the right way to make it scalable is to sync only folders that has been accessed at least one time. I mean : 
- first sync : just sync the first folder tree. It will be instantly ready.
- when the user opens a folder, sync this folder content, but not recursively. And add this folder to the list of folders to keep synced when there are changes in it.
- the user can select manually a folder to be fully synced (like it's already possible)
So the first access of each folder will be a bit slower, but step by step the user will get synced the folders he's using, and will never sync all other folders.

Is this technically possible ? And if yes, what do you (nextcloud devs) think about it ?
It seems to me that it's the actual behaviour of the android desktop client.

**Comment by marcotrevisan** [comment]
I'd like to add that under Mac OS things are changing towards a FileProvider based implementation, which will solve the issue by delegating a good part of the sync logic to MacOS.

IMHO, if under Windows there's no API like FileProvider, then the client should evolve itself to a lazier approach... a "full sync" approach is against scalability and in the long run it's a major limiting factor for a borader adoption of Nextcloud. 
In the case of 500k files and 30 users that are actively working, push notifications tend to generate very frequent peaks of PROPFIND requests coming from all the clients. Such peaks will cause slowdowns not only to the clients themselves but also to the other apps (talk, mail, calendar, deck...), and the end result is a busy server instance that actually is not doing anything except triggering propfinds and responding to propfinds, for files/folders that are often far away from where the actual users are working.
That's why in my hubmle opinion this is a critical and high-priority issue.

**Comment by marcotrevisan** [comment]
@tomdereub I'm in a very similar situation to yours and as a mitigation solution I ended up as follows:
- use a webdav client like Mountain Duck in "online" mode for occasional browsing and work on the folder structure. It has its own issues but it basically works (don't forget to generate an "application" password for this client in the user's Settings -> Security section); 
- also use Nextcloud Client without virtual files, selecting those folders containing the most heavily used projects for the user, and instructing them how to add/remove folders to sync.

In this way, server load is under control (push notifications won't wake up all clients every time) and the clients are snappy enough to work. The advantage is that, for heavily used folders, the NC client has all the files downloaded and ready; the disadvantage is that not all the users are comfortable with such setup. 

Hope it helps

**Comment by tomdereub** [comment]
@marcotrevisan I'm actually trying mountainduck, and it seems to do everything I want with the "smart synchronization" mode. There is an option to index files or not.
So without checking this option, it will not index all files, it will just keep index of visited folders. And there is a option to keep a folder offline on local disk.
So it actually does what nextcloud vfs does, but with 2 advantages (from my point of view) : 
- it's possible not to index all files -» far more scalable
- it mounts the webdav folder as a drive letter, what is usefull (on windows)
It seems that mountainduck has part of it's code opensource, maybe it could be interesting to have a look in it.

**Comment by marcotrevisan** [comment]
Yes, but don't get drunk too fast, it has its own bugs (in Mac OS at least)  :-D 
Avoid unzipping archives in the share for example. Sometimes it'll screw things up, and I don't know why. The safest mode in my experience is the Online mode. If you're in Windows it may behave differently.

**Comment by PhilippSchlesinger** [comment]
Just small addition regarding the initial scan:
Synchronizing placeholder files for an additional 100k files is expected to take 0 seconds (after a previous operation already took over 90 minutes for 60k files):

<img width="285" alt="Screenshot 2023-10-17 101406" src="https://github.com/nextcloud/desktop/assets/2735628/be9e35e9-e966-42f5-89bf-3abe382dab38">

**Comment by tomdereub** [comment]
>  It has been agreed for 2 years now ([#3120 (comment)](https://github.com/nextcloud/desktop/issues/3120#issuecomment-907067592)) without visible progress...

@allexzander @mgallien could you please just give us some idea of the priority of this issue and the ways to solve it ? Like "it's not the priority at the moment, so we don't know when it will be worked on", or "it's very complicated to solve, we have to re-write entirely the sync engine, so it will take some time before we can work on it", or "you're just a few users concerned, so it's not a priority, most of our users don't have so much data"...

As users, we need to know if there is some chance to get VFS scalable at a short or mid term, or if we have to found other solutions. I don't want to see my company giving up with nextcloud and other opensource software we're using, and fall into full microsoft solutions.
I'm trying for some time mountainduck as an alternative, but as @marcotrevisan and @roberix have said, for some cases it's not working as well as nextcloud desktop client. So I need to know a bit more of nextcloud desktop client futur development before deploying it for all users.

**Comment by tomdereub** [comment]
@joshtrichards : you added a label on this issue, what does that mean ? Will somebody start working on it ?

**Comment by OpsecPGR** [comment]
> [@joshtrichards](https://github.com/joshtrichards) : you added a label on this issue, what does that mean ? Will somebody start working on it ?

From what I can see, looks like they began working on this about a week ago.

**Comment by tomdereub** [comment]
This https://github.com/nextcloud/desktop/pull/6461 is exactly what is needed for windows too.

**Comment by PhilippSchlesinger** [comment]
Dear Nextcloud developers, @allexzander
It would be great if you could shed some light on what is actually being worked on. Many are following this bug and many of us contributed to this issue.

See https://github.com/nextcloud/desktop/issues/4918 for a description of a performance problem (PR intended to solve the problem in https://github.com/nextcloud/desktop/pull/5941) with the tray window. Solving this heavy issue could also pay off in improving the speed problems with initial sync.

**Comment by psxvoid** [comment]
For me, the initial sync is in progress for several days, and seems like laptop restarts, network connection issues are restarting this process from scratch each time. On the screenshot the number of total files is constantly increasing, and notice, file synced count is always 0:
<img width="346" alt="image" src="https://github.com/user-attachments/assets/ff584544-a4bd-49fb-81b4-902f82586682">

and there are no any files in the sync folder except those:
<img width="505" alt="image" src="https://github.com/user-attachments/assets/2c267554-dace-40a0-b078-fd3941718485">

**Comment by tomdereub** [comment]
This first step of initial sync is very hard on the server. You can have a look of cpu consumption of your server, I think it's the bottleneck : in my case I have an intel i5-10210U, 6 cores dedicated to my server, and it's using almost 100% of all cores while doing this first scan of all files. I have about 700 000 files, and it takes between 1/2h and 1h to make the scan. So I'm not surprised that it takes so long on a RPi.
Once the server side scan is finished, it takes up to 48h non stop on the client to create the whole file tree.
In my case, once the first sync is done, it's working well (20 persons using it), and the load on the server is ok.
Looking forward to some improvement on this issue...

**Comment by ne0YT** [comment]
@Rello hey there, do you have an estimation when this will be done ?

we are planning to move from our weird software-solution built on top of windows builtin webdav which has a lot of other issues and will be canceled.. so a switch will be needed as fast as possible.

**Comment by vagner-dias** [comment]
OneDrive uses a more efficient approach by downloading the file and folder structure from the server first and immediately replicating this structure in the local filesystem. This approach appears to be more effective than Nextcloud's current method, which downloads the entire file/folder structure first and only begins creating it locally afterward.

**Comment by ne0YT** [comment]
@Rello hey there, what's up? the windows client isn't going to be enhanced anymore?

**Comment by Rello** [comment]
Hello @ne0YT as the topics is not closed, its not rejected 🤷🏼‍♂️
just because we are reassigning topics from tracking boards does not give a feedback on community requests

**Comment by ne0YT** [comment]
@Rello my concerns stems from the fact that this is open since 2.5 years and in my opinion it's an extremely relevant issue.
as there is no one currently assigned it feels like it might never happen.
so when it gets removed from the tracking board it seems that the possibility it gets assigned to someone lowers.
given the initially brought up fact and that there are two other issues related to similar issues where one was closed:
https://github.com/nextcloud/desktop/pull/5780#issuecomment-2558119795
and the other one was also removed from the tracking board (3.5 years old):
https://github.com/nextcloud/desktop/issues/3120#issuecomment-2558057624

I just assume it will not happen any time soon.

**Comment by Rello** [comment]
Hello,
I think there is quite some misunderstanding.
Topics might be related to bigger initiatives. not everything is on public tickets or planing boards

**Comment by ne0YT** [comment]
> Hello, I think there is quite some misunderstanding. Topics might be related to bigger initiatives. not everything is on public tickets or planing boards

thank you for clarifying.
Is there some kind of an estimate or a due -date?

**Comment by ne0YT** [comment]
@Rello any idea on this?

**Comment by ne0YT** [comment]
@Rello please clarify this for us:
"Topics might be related to bigger initiatives. not everything is on public tickets or planing boards"

what is the case here and is there any ETA or do we need to find another solution?

**Comment by r0ckychan** [comment]
Just wondering. Is it possible to only sync the database rather than syncing the folder structure via file system tructure?

**Comment by ne0YT** [comment]
@Rello please any new insider- insides about this?

**Comment by Styx13** [comment]
Hello,

I recently added 250 000 files to my nextcloud (which already had around 50 000 files) and I am having the same issue described by others here: it the sync is extremely slow. It goes fairly fast for the few first hundreds, but then when it get to 2000  it starts slowing down, at 10000 or so it's roughly doing 1 per second, it looks like it is going to take days to sync.

Windows 11 (24H2), Nextcloud client 3.16.5, nextcloud server 30.0.8
Using virtual filesystem.

Nextcloud is great, which is why I start putting more and more in it, and the virtual filesystem is extremely useful when the nextcloud content is several terabytes large, however this initial sync being extremely slow is a big issue.

**Comment by tomdereub** [comment]
I'm facing this issue again while updating the client to 3.16.5 : it seems it's doing a new "first sync", or something like that : https://github.com/nextcloud/desktop/issues/8349
With 20 pc upgrading the client at the same time, and about 1,5 million files, the server is at 100% CPU (7 cores i5-10210U CPU @ 1.60GHz) since this morning, so during 10 hours...
I'm wondering : what is the server doing with so much CPU ? "Just" scanning and comparing files ? One by one, for each client ?
This issue is between server and client, is it relevant to open it on server repo too ? I haven't found any similar issue on it.

**Comment by Styx13** [comment]
On my side, I noticed that one of my client was syncing much faster than the other one.
And I found the difference was the slower client was running with nextcloud client 3.16.5  and the faster one is running with 3.16.4.

I then downgraded the slow client from 3.16.5 to 3.16.4  and it started syncing much faster.

For comparison, with the 3.16.5 client it was syncing at around 1 file per second (after reaching around 15000 files). With the 3.16.4 client, it was syncing around 100 files per second for the whole duration. So syncing 250K files took around 40 minutes with the 3.16.4 client, while with the 3.16.5 it would probably have taken a whole week.

So there is definitively a performance regression with VFS syncing in the client 3.16.5.

And even without this performance regression, 40 minutes to sync 250K files from a VFS is still fairly long, it would mean that it would take around 2h40m to sync 1M files.

**Comment by ne0YT** [comment]
Please, this is the one issue keeping us from rolling out Nextcloud with the official client!
Before we were using the software with a custom made client which was using the built-in windows WebDAV client which is officially not supported anymore so we decided to stop deploying more Nextcloud installations.
The built-in client already sucked enough.. but it was possible to get clients ready ultra fast.

So please there needs to be a fix or we are forced to completely abandon Nextcloud for any customer needing a Virtual Drive Feature (which for us is everyone)

Is there anything we can do to make this happen?
Can we do a crowd sourcing event of some sorts?
I tried to find people being Able to work on this but sadly didn't succeed.
Maybe someone have a heads up for me in this matter?

**Comment by hodyroff** [comment]
https://github.com/nextcloud/desktop/issues/8349 made this even slower.
ownCloud/OpenCloud/Seafile will only bring you so far ...

Why not use rclone for Windows - if you don't need offline sync this can be a very fast option ... @ne0YT

**Comment by ne0YT** [comment]
> [#8349](https://github.com/nextcloud/desktop/issues/8349) made this even slower. ownCloud/OpenCloud/Seafile will only bring you so far ...
> 
> Why not use rclone for Windows - if you don't need offline sync this can be a very fast option ... [@ne0YT](https://github.com/ne0YT)

Do you have a well working config for rclone under windows?
How does it behave with local storage usage etc. / did you use it yourself for some time?

**Comment by hodyroff** [comment]
> > [#8349](https://github.com/nextcloud/desktop/issues/8349) made this even slower. ownCloud/OpenCloud/Seafile will only bring you so far ...
> > Why not use rclone for Windows - if you don't need offline sync this can be a very fast option ... [@ne0YT](https://github.com/ne0YT)
> 
> Do you have a well working config for rclone under windows? How does it behave with local storage usage etc. / did you use it yourself for some time?

I am using it on Linux a lot, not under Windows personally.
Check out this recent post which tells me that generally there are plenty of options and it could be a great choice under Windows as well. Just that your backend would be Nextcloud instead of oneDrive ... but that part should be transparent to the usage and you can adopt easily to the proper config.
https://blog.ligos.net/2025-05-30/Connecting-To-OneDrive-With-RClone.html
Next in line could be Cyberduck/Mountainduck ...

Of course you need to find a way to roll this out to the Windows users with GPO/Intunes etc. or you use the opportunity and turn them into Linux Desktop users!

The type of ask in this issue is not likely to show up anytime soon, thats the other message I like to clearly relay. It is going the other direction currently ... therefor a WebDAV mount and browing option makes more sense and yes the build into Windows one is a nightmare and Microsoft will not touch it or fix it.

**Comment by ne0YT** [comment]
> > > [#8349](https://github.com/nextcloud/desktop/issues/8349) made this even slower. ownCloud/OpenCloud/Seafile will only bring you so far ...
> > > Why not use rclone for Windows - if you don't need offline sync this can be a very fast option ... [@ne0YT](https://github.com/ne0YT)
> > 
> > 
> > Do you have a well working config for rclone under windows? How does it behave with local storage usage etc. / did you use it yourself for some time?
> 
> I am using it on Linux a lot, not under Windows personally. Check out this recent post which tells me that generally there are plenty of options and it could be a great choice under Windows as well. Just that your backend would be Nextcloud instead of oneDrive ... but that part should be transparent to the usage and you can adopt easily to the proper config. https://blog.ligos.net/2025-05-30/Connecting-To-OneDrive-With-RClone.html Next in line could be Cyberduck/Mountainduck ...
> 
> Of course you need to find a way to roll this out to the Windows users with GPO/Intunes etc. or you use the opportunity and turn them into Linux Desktop users!
> 
> The type of ask in this issue is not likely to show up anytime soon, that's the other message I like to clearly relay. It is going the other direction currently ... therefore a WebDAV mount and browing option makes more sense and yes the build into Windows one is a nightmare and Microsoft will not touch it or fix it.

Well.. I was asking for a good working config for Windows as I had major issues as we tried that before too when building our helper app which worked with the windows WebDAV client (and having that same feature as the official Client of using app-passwords by logging in to nextclouds web once..) Sadly I didn't really document the outcome but it was so bad that we chose the built-in client over it.
But I can retry later today or tomorrow what was the issue.
the thing is also that we modified the context menu of nextcloud to also feature an "open - locally" feature similar to nextclouds own implementation but supported it as a standard-action for pre defined file types. And smh this modification also stopped working some versions ago when they changed the way the UI works (we just injected js to the user-interface serverside for the mod.)

maybe someone has an idea on how to do this nowadays too? (Changing standard-action and modifying the context-menu)

Generally I would really appreciate things just working with the official client too..

**Comment by Rello** [comment]
Hello,

there is progress ongoing to make the sync quicker. but the discussion here is also comparing apples and oranges a little bit.

@hodyroff your linked ticket is just a one-time-action. When a security relevant parameter changes for all files, they need to be synced. This is the equivalent of using rclone - but changing all change-timestamps for every file on the server. 

rclone can also not compared to the sync client because if only dump-syncing file - then yes. this can be done with this in quick. The client does much more like file locking, E2E, sharing, ...

The 3.17 macOS VFS implementation already got the improvement for "lazy loading" to only read the folders once navigated and keep them monitored in the working set. This is a similar approach that is being evaluated for Windows too.

...updates to follow...

**Comment by ne0YT** [comment]
> This is a similar approach that is being evaluated for Windows too

Is there any ETA for this?

**Comment by ne0YT** [comment]
> > > [#8349](https://github.com/nextcloud/desktop/issues/8349) made this even slower. ownCloud/OpenCloud/Seafile will only bring you so far ...
> > > Why not use rclone for Windows - if you don't need offline sync this can be a very fast option ... [@ne0YT](https://github.com/ne0YT)
> > 
> > 
> > Do you have a well working config for rclone under windows? How does it behave with local storage usage etc. / did you use it yourself for some time?
> 
> I am using it on Linux a lot, not under Windows personally. Check out this recent post which tells me that generally there are plenty of options and it could be a great choice under Windows as well. Just that your backend would be Nextcloud instead of oneDrive ... but that part should be transparent to the usage and you can adopt easily to the proper config. https://blog.ligos.net/2025-05-30/Connecting-To-OneDrive-With-RClone.html Next in line could be Cyberduck/Mountainduck ...
> 
> Of course you need to find a way to roll this out to the Windows users with GPO/Intunes etc. or you use the opportunity and turn them into Linux Desktop users!
> 
> The type of ask in this issue is not likely to show up anytime soon, thats the other message I like to clearly relay. It is going the other direction currently ... therefor a WebDAV mount and browing option makes more sense and yes the build into Windows one is a nightmare and Microsoft will not touch it or fix it.

rclone on windows using nextcloud is a nightmare. it connects immidiately but loading content of files is slow, deleting is slow.. files are not up to date etc.

**Comment by camilasan** [comment]
> Please, this is the one issue keeping us from rolling out Nextcloud with the official client!
> Is there anything we can do to make this happen?

If you are a customer, please submit a support ticket for this. We can then help you more closely.

**Comment by ne0YT** [comment]
> > Please, this is the one issue keeping us from rolling out Nextcloud with the official client!
> > Is there anything we can do to make this happen?
> 
> If you are a customer, please submit a support ticket for this. We can then help you more closely.

Sadly that is currently not a possibility as we're on a net-loss for everything Nextcloud-related.
Besides the technical downsides in the Client-Software we invest quite some time to think around customers needs to somehow make it possible to use the Software in the first place.
If it's possible to participate in a fund-raiser for a specific issue or feature I'm glad to join.

**Comment by Rello** [comment]
https://github.com/nextcloud/desktop/pull/8486

one improvement on the speed up of the propagation phase. more to come...

**Comment by tomdereub** [comment]
> The 3.17 macOS VFS implementation already got the improvement for "lazy loading" to only read the folders once navigated and keep them monitored in the working set. This is a similar approach that is being evaluated for Windows too.
> 
> ...updates to follow...

Any news about that ? Lazy loading is exactly what we need, with about 1 million file synced with VFS, it would resolve quite a lot of issues : 
- speed up initial sync
- speed up sync
- reduce load on client, nextcloud is consuming lot of cpu to keep all files synced with 30 persons working on it every day, and it has a significant impact on computers (battery consumption and slowing other usages)
- maybe reduce load on the server too ?
- enable nextcloud to be really scalable, with almost no more limit on the number of files it can manage with the client

**Comment by mgallien** [comment]
more work is needed to finish https://github.com/nextcloud/desktop/pull/8359
this will eventually come to a release

**Comment by VLNickR** [comment]
Has this been resolved yet?

We have a client with more than 1m Files split between their team folders and want to have them all synced.

We have them syncing them as individual folders but then shared folders wont appear automatically and need to be synced individually now.

**Comment by VLNickR** [comment]
I see there is a working version of the enhancements on the latest daily build, 3.5tb and 1.2 million files synced in just a few seconds.

**Comment by Oisins** [comment]
> I see there is a working version of the enhancements on the latest daily build, 3.5tb and 1.2 million files synced in just a few seconds.

Do you know when it will be released or which commit it was?

**Comment by limatus** [comment]
> more work is needed to finish [#8359](https://github.com/nextcloud/desktop/pull/8359) this will eventually come to a release

I've just tested this with a daily build. The improvement is immense and also helps with other issues (client not able to delete folders within a very long path). A job that took days before was done within a second. @mgallien that's a remarkable job, any plan when you think to implement it to the stable releases? Thanks to the team for the work!

**Comment by VLNickR** [comment]
> > I see there is a working version of the enhancements on the latest daily build, 3.5tb and 1.2 million files synced in just a few seconds.
> 
> Do you know when it will be released or which commit it was?

I'm just running the Daily build as a test on a few lab machines.
not sure on the commit sorry

**Comment by MaxXor** [comment]
> more work is needed to finish [#8359](https://github.com/nextcloud/desktop/pull/8359) this will eventually come to a release

Thanks, is this included in latest version v4.0.4? 🙂

**Comment by limatus** [comment]
> > more work is needed to finish [#8359](https://github.com/nextcloud/desktop/pull/8359) this will eventually come to a release
> 
> Thanks, is this included in latest version v4.0.4? 🙂

It's not, it is only part of the alpha versions/daily versions of the client, you can download here: https://download.nextcloud.com/desktop/daily/

Or set your update canal to alpha.

It still saves hours of work ;)

**Comment by noremac09** [comment]
Switching to the daily build significantly improved the experience for me. I had a sync of ~400k files stuck for nearly three hours, and the Nextcloud client was behaving very poorly during that time.

After moving to the daily version, it picked everything up almost immediately and created all VFS virtual files without issues.

**Comment by moose999** [comment]
Daily build seems excellent! I just sync'd 8+ million files instantaneously. Very much looking forward to the stable 👍

**Comment by tomdereub** [comment]
@noremac09 @moose999 : did you test with a fresh installation ? Or with an upgrade of a client already using VFS ?

**Comment by moose999** [comment]
@tomdereub 
Upgrade of a client already using VFS - although the initial sync had _never_ completed

**Comment by johnjaylward** [comment]
I am seeing absolutely horrible performance using client 33.0.2 on windows 10. I did a fresh install of the client, and new sync and it was able to get my only my first level directories. Upon retrieving the first level directories, NextCloud gave me a message that the sync was complete. However, if I try to navigate into any of the directories I get an error message:

<img width="475" height="163" alt="Image" src="https://github.com/user-attachments/assets/9cb55f45-f162-4c62-aa1f-99bf9c1e2b02" />

If I try to use the actual Nextcloud client I get this message:

<img width="349" height="192" alt="Image" src="https://github.com/user-attachments/assets/faaf8beb-7e96-4f97-9917-e00685e8c181" />

I do periodically see windows update some of my folders with the little empty cloud icon, but alas, it's unusable

<img width="550" height="25" alt="Image" src="https://github.com/user-attachments/assets/1277f3a6-c0e7-4f68-b7c5-72f2cad6bcbf" />


Using Client version 33.1.0 alpha, I get stuck not even getting the top level directories

<img width="353" height="513" alt="Image" src="https://github.com/user-attachments/assets/61c7b215-ea7c-4159-bbbc-3137ef5d1002" />

I'd also like to add that this locks up file explorer. I'm unable to open new file explorer windows once Nextcloud locks up like this.

**Comment by djamps** [comment]
> I'd also like to add that this locks up file explorer. I'm unable to open new file explorer windows once Nextcloud locks up like this.

I'm getting the exact behavior on 33.x.   However even 4.x does the same thing.   Had to go all the way back to 3.10 for the initial virtual sync to fire off successfully without locking up on "Preparing for sync".

**Comment by mgallien** [comment]
people, please keep issues on topic
if you want to report a bug, please create a new one instead of making off-topic comments in a *enhancement* request that is unrelated

**Comment by mgallien** [comment]
by the way, I think this was wrongly kept open
folders and files are now populated on-demand on Windows when the user needs them

**Comment by Zegorax** [comment]
@mgallien This is issue is not windows-specific. The initial sync performance on Linux is still horrible and takes ages to sync a 3TB teamfolder for new clients (or even existing ones that needs a full resync for some reason)

**Comment by beedell-roke** [comment]
@Zegorax, except AOSP and Darwin, non-Windows clients don't support Cloud Files API equivalents, so their files are not virtual.

**Comment by johnjaylward** [comment]
@mgallien I fail to see how my comment was off topic. This is 100% on topic for horrible sync performance when using virtual files on windows 10.

The issue still exists. I'm not able to turn on VFS on any recent version of nextcloud client.

**Comment by Zegorax** [comment]
@roke-julian-lockhart This feature is still called VFS on all platforms, regardless of whether they call the CF API or extension-VFS.

I've read the history of this issue again, and I did not find anything specifying that it's for Windows-only.

I'm subscribed to this issue since at least 2023 waiting for a possible fix. Do we truly need to reopen a new one for the same problem ?
